### PR TITLE
Feature/smar 40

### PR DESCRIPTION
**Pull request by RaznoOleg**
Description changes:
• Created a forgot password form;
• Added a link to the Forgot password form on the Login form, made, that link change color when hovering over it;
• Added a link to the Login form on the Forgot password form, made, that link change color when hovering over it;
• Made, that the “Submit” button change color and become clickable if there are no errors;
• Created a confirmation form;
• Made, that when clicking the "Submit" button open a confirmation form;
• Added, that when clicking on the logo, open the login form. 

### General
- [x] Assigned myself to the PR
- [x] Assigned the appropriate labels to the PR
- [x] Assigned the appropriate reviewers to the PR
- [ ] Updated the documentation
- [x] Performed a self-review of my code
- [x] Don't have "any" on my code
- [x] Don't have magic numbers
- [x] Compare only with constants not with strings
- [x] No ternary operator inside ternary operator
- [x] Don't have commented code
- [x] All links in on ENV files
- [x] Used camelCase for variables and functions
- [x] Functions is public only if it's used outside the class
- [x] No hardcoded values
- [x] Check your commit messages meet the [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/).

### Frontend
- [x] Components and business logic are separated
- [x] Colors, Font Size, and Font Name is on the theme or in the constants
- [x] No text in the components, use i18n approach
- [x] No inline styles
- [x] Imports are absolute
- [x] Attach a screenshot if PR has visual changes.

#### Forgot password form:
image

#### Link to Forgot password form in Login form:
image

#### Link to Login form in Forgot password form:
image

#### Сhanging color and becoming a clickable button "Submit" when there are no errors:
image
image

#### Confirmation form:
image

#### Сlicking on the logo and opening the login form:
image
image

### CODE REVIEW

File: src/api/auth/auth.api.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    gender: string;
+    country: string;
+    city: string;
+    birthDate: string;
+    address: string;
+    timeZone: string;
+}
+
+export const authAPI = {
+    signUp({firstName, lastName, email, phoneNumber, password, confirmPassword, role, specialization, gender, country, city, birthDate, address, timeZone,}: AuthSignUpDto) {
```
Comment (lyudad) [inline src/api/auth/auth.api.ts]: `{firstName, lastName, email, phoneNumber, password, confirmPassword, role, specialization, gender, country, city, birthDate, address, timeZone,}` what is it? change

File: src/components/Auth/ResetPasswordForm/index.tsx [hunk 2]
```diff
@@ -22,93 +22,94 @@ import { confirmPassword, end, password } from '@constants/auth';
 import { resetPasswordSchema } from '@validation/auth.validate';
 
 function ResetPasswordForm() {
-  const [showPassword, setShowPassword] = React.useState(false);
-  const [showConfirmPassword, setShowConfirmPassword] = React.useState(false);
+    const [showPassword, setShowPassword] = React.useState(false);
```
Comment (lyudad) [inline src/components/Auth/ResetPasswordForm/index.tsx]: const [showPassword, setShowPassword] = React.useState<boolean>(false);

File: src/components/Auth/SignUpSecondForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+    },
+
+    resolver: yupResolver(signUpSecondStepSchema)
+  });
+
+  const onSubmit = (data: ISignUp) => {
+    data.specialization = Number(data.specialization);
+    const combinedObj = Object.assign({}, dataSignUpFirst, data);
+
+    dispatch(signUpQuery(combinedObj as AuthSignUpDto)).then((res: IResponse) => {
```
Comment (lyudad) [inline src/components/Auth/SignUpSecondForm/index.tsx]: `as` delete

File: src/components/Input/index.tsx [hunk 4]
```diff
@@ -1,11 +1,10 @@
 import {Control, Controller} from "react-hook-form";
 import { TextFieldProps } from "@mui/material/TextField/TextField";
 import {TextField} from "@mui/material";
-import { SignUpFields } from '@types';
-import { LoginFields } from '@types';
 import {InputContainer} from "@components/Input/styles";
-import { ISignUp } from "@components/Auth/type";
-import { ILogin } from "@components/Auth/type";
+import {ISignUp} from "@components/Auth/type";
+import { SignUpFields } from "types/auth.type";
```
Comment (lyudad) [inline src/components/Input/index.tsx]: you have a problem with pritier, check line 5 and 6 and find 10 differences

File: src/redux/slices/auth/signUp/index.ts [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+  role: '',
+  specialization: 0,
+  gender: '',
+  country: '',
+  city: '',
+  birthDate: '',
+  address: '',
+  timeZone: '',
+  isLoading: false,
+  token: JSON.parse(<string>localStorage.getItem('token')),
```
Comment (lyudad) [inline src/redux/slices/auth/signUp/index.ts]: no

File: src/redux/slices/auth/signUp/index.ts [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+  error: null,
+};
+
+
+export const signUpQuery: any = createAsyncThunk(
+    'signUp/signUpQuery',
+    async (data: AuthSignUpDto, { rejectWithValue }) => {
+      try {
+        return await authAPI.signUp(data);
+      } catch (err: any) {
```
Comment (lyudad) [inline src/redux/slices/auth/signUp/index.ts]: go through check list

File: src/redux/slices/auth/signUp/index.ts [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+      state.confirmPassword = action.payload.confirmPassword;
+    },
+  },
+  extraReducers: {
+    [signUpQuery.pending.type]: (state) => {
+      state.isLoading = true;
+    },
+    [signUpQuery.fulfilled.type]: (state, action) => {
+      state.token = action.payload.accessToken;
+      localStorage.setItem('token', JSON.stringify(state.token));
```
Comment (lyudad) [inline src/redux/slices/auth/signUp/index.ts]: no

File: src/api/auth/auth.api.ts (hunk 1 above)
Comment (RaznoOleg) [reply src/api/auth/auth.api.ts]: resolved

File: src/components/Auth/ResetPasswordForm/index.tsx (hunk 2 above)
Comment (RaznoOleg) [reply src/components/Auth/ResetPasswordForm/index.tsx]: resolved

File: src/components/Auth/SignUpSecondForm/index.tsx (hunk 3 above)
Comment (RaznoOleg) [reply src/components/Auth/SignUpSecondForm/index.tsx]: resolved

File: src/components/Input/index.tsx (hunk 4 above)
Comment (RaznoOleg) [reply src/components/Input/index.tsx]: resolved

File: src/redux/slices/auth/signUp/index.ts (hunk 5 above)
Comment (RaznoOleg) [reply src/redux/slices/auth/signUp/index.ts]: resolved with redux-persist

File: src/redux/slices/auth/signUp/index.ts (hunk 6 above)
Comment (RaznoOleg) [reply src/redux/slices/auth/signUp/index.ts]: resolved

File: src/redux/slices/auth/signUp/index.ts (hunk 7 above)
Comment (RaznoOleg) [reply src/redux/slices/auth/signUp/index.ts]: resolved with redux-persist

File: src/components/Auth/SignUpFirstForm/index.tsx [hunk 8]
```diff
@@ -1,46 +1,59 @@
-import React, {useEffect} from 'react';
-import {useTranslation} from "react-i18next";
-import {useForm} from "react-hook-form";
-import {yupResolver} from "@hookform/resolvers/yup";
-import { useGoogleLogin } from '@react-oauth/google';
-import {IconButton, InputAdornment } from '@mui/material';
+import React, { useEffect, useState } from 'react';
```
Comment (lyudad) [inline src/components/Auth/SignUpFirstForm/index.tsx]: SignUpFirstForm why there is first and not first form. rename or make it more generic

File: src/validation/auth.validate.ts [hunk 9]
```diff
@@ -1,66 +1,86 @@
...
     .oneOf([yup.ref('password')], '*Passwords does not match'),
 });
+export const signUpSecondStepSchema = yup.object().shape({
+  address: yup.string().required('Field is required'),
+  city: yup.string().required('Field is required'),
+  country: yup.string().required('Field is required'),
+  gender: yup.string().required('Field is required'),
+  specialization: yup.string().required('Field is required'),
+  role: yup.string().required('Field is required'),
+  timeZone: yup.string().required('Field is required'),
```
Comment (lyudad) [inline src/validation/auth.validate.ts]: better with translator file

Comment (RaznoOleg) [reply src/validation/auth.validate.ts]: improved

File: src/components/Auth/SignUpFirstForm/index.tsx (hunk 8 above)
Comment (RaznoOleg) [reply src/components/Auth/SignUpFirstForm/index.tsx]: In our project we have two registration steps. Therefore, for this we created two components - "SignUpFirstStepForm" and "SignUpSecondStepForm". I also put these two components in a shared folder "SignUpForm" for convenience:
image

**Pull request merged**